module-system/module-implementations/sov-attester-incentives/src/call.rs

```diff
@@ -0,0 +1,764 @@
+            self.bonded_challengers
+                .remove(context.sender(), working_set);
+
+            // Emit the unbonding event
+            working_set.add_event(
+                "unbonded_challenger",
+                &format!("amount_withdrawn: {old_balance:?}"),
+            );
+        }
+
+        Ok(CallResponse::default())
+    }
+
+    /// The attester starts the first phase of the two phase unbonding. We put the current max
+    /// finalized height with the attester address in the set of unbonding attesters iff the attester
+    /// is already present in the unbonding set
+    pub(crate) fn begin_unbond_attester(
+        &self,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<sov_modules_api::CallResponse, AttesterIncentiveErrors> {
+        // First get the bonded attester
+        if let Some(bond) = self.bonded_attesters.get(context.sender(), working_set) {
+            let finalized_height = self
+                .light_client_finalized_height
+                .get(working_set)
+                .expect("Must be set at genesis");
+
+            // Remove the attester from the bonding set
+            self.bonded_attesters.remove(context.sender(), working_set);
+
+            // Then add the bonded attester to the unbonding set, with the current finalized height
+            self.unbonding_attesters.set(
+                context.sender(),
+                &UnbondingInfo {
+                    unbonding_initiated_height: finalized_height,
+                    amount: bond,
+                },
+                working_set,
+            );
+        }
+
+        Ok(CallResponse::default())
+    }
+
+    pub(crate) fn end_unbond_attester(
+        &self,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<sov_modules_api::CallResponse, AttesterIncentiveErrors> {
+        // We have to ensure that the attester is unbonding, and that the unbonding transaction
+        // occurred at least `finality_period` blocks ago to let the attester unbond
+        if let Some(unbonding_info) = self.unbonding_attesters.get(context.sender(), working_set) {
+            // These two constants should always be set beforehand, hence we can panic if they're not set
+            let curr_height = self
+                .light_client_finalized_height
+                .get(working_set)
+                .expect("Should be defined at genesis");
+            let finality_period = self
+                .rollup_finality_period
+                .get(working_set)
+                .expect("Should be defined at genesis");
+
+            if unbonding_info.unbonding_initiated_height + finality_period > curr_height {
+                return Err(self.slash_burn_reward(
+                    context.sender(),
+                    Role::Attester,
+                    SlashingReason::UnbondingNotFinalized,
+                    working_set,
+                ));
```

Comment:
I don't think we need to slash here - we can just revert

Reply:
Fixed